exercises/all-your-base/src/example.c

```diff
@@ -0,0 +1,37 @@
+#include "all_your_base.h"
+#include <math.h>
+
+size_t rebase(int16_t digits[DIGITS_ARRAY_SIZE], int16_t fromBase,
+              int16_t toBase, size_t numDigits)
+{
+   uint16_t denary = 0;
+   size_t newNumDigits = 0;
+
+   if ((fromBase <= 1) || (toBase <= 1) || (numDigits <= 0))
+      return 0;                 /* invalid bases or length */
+   if (digits[DIGITS_ARRAY_SIZE - numDigits] == 0)
+      return 0;                 /* leading zeros */
+
+   /* convert to denary */
+   for (size_t i = DIGITS_ARRAY_SIZE - 1, j = 0;
+        i >= DIGITS_ARRAY_SIZE - numDigits; --i, ++j) {
+      if (digits[i] < 0)
+         return 0;              /* negative digits */
+      if (digits[i] >= fromBase)
+         return 0;              /* invaid positive digit */
+      denary += digits[i] * pow(fromBase, j);
+   }
+
+   /* calculate and store new digits */
+   for (size_t i = DIGITS_ARRAY_SIZE - 1; denary > 0; --i) {
+      digits[i] = denary % toBase;
+      denary /= toBase;
+      ++newNumDigits;
+   }
+
+   /* ensure rest of array is zero */
+   for (int i = DIGITS_ARRAY_SIZE - (newNumDigits + 1); i >= 0; --i)
+      digits[i] = 0;
```

Comment:
Yeah `memset` will work because while it's in `string.h`, it's really just filling things character by character. In order to zero out an entire `int` array, you just have to make sure that you provide it the length, in bytes, of the entire `int` array (with appropriate use of `sizeof`) and it will dutifully zero out every single byte. It doesn't know the type of the array (or even whether it's an array!) and views it as just a big block of memory.